### ISSUE TITLE
Migration from 0.16.2 to 1.1 troubleshooting

### ISSUE THREAD

**Issue by jackalsh** [jenkinsci/docker-plugin#572]
We are upgrading docker-plugin from version 0.16.2 to 1.1
After upgrade, all current container template configurations from Manage Jenkins -> Configure System just disappeared. Can you please tell me if this is a known issue because of some back incompatibility?
Thanks in advance.

**Comment by ndeloof** [comment]
this is not expected, data should have been converted on upgrade.
Will need to prepare a test environment to reproduce

**Comment by CedricGatay** [comment]
I faced an issue too when upgrading from 1.0.4 to 1.1.1 where the checkbox "Commit agent's Docker container" has been automatically selected on all jobs, even if it was not selected previously and with an empty tag suffix. 

It leads to a crash at the end of the job with a message like this "com.github.dockerjava.api.exception.InternalServerErrorException: invalid reference format: repository name must be lowercase"

The following groovy script run in the console script page sorted everything out:

```
import hudson.matrix.*
import jenkins.model.*;
import com.cloudbees.hudson.plugins.folder.*

Jenkins.getInstance().getAllItems().each {
if (!(it instanceof MatrixConfiguration)) {
  try{
    def dockerPlugin = it.properties.find { it.key.class == com.nirima.jenkins.plugins.docker.DockerJobProperty.DescriptorImpl }?.value
    if ("".equals(dockerPlugin?.additionalTag)){
        println "Removing from ${it.name}"
        it.removeProperty(dockerPlugin)
        it.save()
    }
  }catch(e){
    println "Unable to check job ${it.name}, ${e.getMessage()}"
  }
}
}
```

To prevent polluting the issue tracker with a new issue I though it was a better place to document this here. Please forgive me if I did a mistake.

**Comment by jeritiana** [comment]
From @CedricGatay's inputs, as I have access to all Jenkins data, I unchecked the "Commit agent's Docker container" checkbox directly by editing the `config.xml` file in each job's directory and then restarted Jenkins.

The following command allows to do so (in the `jobs` directory under `JENKINS_HOME` e.g.: `/var/jenkins_home/jobs`):
```
find . -name config.xml -exec sed -i '/<com.nirima.jenkins.plugins.docker.DockerJobProperty/{ :x; N; /<\/com.nirima.jenkins.plugins.docker.DockerJobProperty/!bx; d; }' {} \;
```

Then restart Jenkins.

It is not really pretty, but that's the fastest way I found to handle all jobs and keep data consistency to be able to replicate to another Jenkins instance.
